File: backend-paf/src/main/java/com/example/paf/service/WorkplanService.java
package com.example.paf.service;

import com.example.paf.model.Workplan;
import com.example.paf.repository.WorkplanRepository;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

import java.util.List;
import java.util.Optional;

@Service
public class WorkplanService {

    @Autowired
    private WorkplanRepository workplanRepository;

    public List<Workplan> getAllWorkplans() {
        return workplanRepository.findAll();
    }

    public Optional<Workplan> getWorkplanById(String id) {
        return workplanRepository.findById(id);
    }

    public List<Workplan> getWorkplansByUser(Long userId) {
        return workplanRepository.findByCreatedBy(userId);
    }

    public Workplan createWorkplan(Workplan workplan) {
        return workplanRepository.save(workplan);
    }

    public Optional<Workplan> updateWorkplan(String id, Workplan workplanDetails) {
        return workplanRepository.findById(id)
                .map(existingWorkplan -> {
                    existingWorkplan.setTitle(workplanDetails.getTitle());
                    existingWorkplan.setDescription(workplanDetails.getDescription());
                    existingWorkplan.setStartDate(workplanDetails.getStartDate());
                    existingWorkplan.setEndDate(workplanDetails.getEndDate());
                    existingWorkplan.setCompleted(workplanDetails.isCompleted());
                    return workplanRepository.save(existingWorkplan);
                });
    }

    public boolean deleteWorkplan(String id) {
        return workplanRepository.findById(id)
                .map(workplan -> {
                    workplanRepository.delete(workplan);
                    return true;
                })
                .orElse(false);
    }

    public List<Workplan> getCompletedWorkplans() {
        return workplanRepository.findByCompletedTrue();
    }

    public List<Workplan> getPendingWorkplans() {
        return workplanRepository.findByCompletedFalse();
    }

    public Optional<Workplan> markWorkplanAsCompleted(String id) {
        return workplanRepository.findById(id)
                .map(workplan -> {
                    workplan.setCompleted(true);
                    return workplanRepository.save(workplan);
                });
    }
}

File: workplan-frontend/src/services/workplanService.js
import axios from 'axios';

const API_URL = 'http://localhost:8080/api/workplans'; // Ensure this matches the backend URL

export const getAllWorkplans = () => axios.get(API_URL);
export const getWorkplanById = (id) => axios.get(`${API_URL}/${id}`);
export const createWorkplan = (workplan) => axios.post(API_URL, workplan);
export const updateWorkplan = (id, workplan) => axios.put(`${API_URL}/${id}`, workplan);
export const deleteWorkplan = (id) => axios.delete(`${API_URL}/${id}`);
export const getCompletedWorkplans = () => axios.get(`${API_URL}/completed`);
export const getPendingWorkplans = () => axios.get(`${API_URL}/pending`);
export const markWorkplanAsCompleted = (id) => axios.patch(`${API_URL}/${id}/complete`);
